crates/store/src/db/models/queries/accounts.rs

```diff
@@ -843,10 +808,8 @@ pub(crate) fn upsert_accounts(
             },
 
             AccountUpdateDetails::Delta(delta) => {
-                let mut rows = select_details_stmt(conn, account_id)?.into_iter();
-                let Some(account_before) = rows.next() else {
-                    return Err(DatabaseError::AccountNotFoundInDb(account_id));
-                };
+                // Reconstruct the full account from database tables
+                let account = reconstruct_full_account_from_db(conn, account_id)?;
```

Comment:
This is a temporary solution, right? In the longer term, we can't read the full account from DB update it, and then save it back. Let's add a TODO here and add this to the list of follow-ups we need to do.